plugins/modules/system/vdo.py

```diff
@@ -360,69 +358,62 @@ def list_running_vdos(module, vdocmd):
 #
 # @return vdocmdoptions  A string to be used in a 'vdo <action>' command.
 def start_vdo(module, vdoname, vdocmd):
-    rc, out, err = module.run_command("%s start --name=%s" % (vdocmd, vdoname))
+    rc, out, err = module.run_command([vdocmd, "start", "--name=%s" % vdoname])
     if rc == 0:
         module.log("started VDO volume %s" % vdoname)
-
     return rc
 
 
 def stop_vdo(module, vdoname, vdocmd):
-    rc, out, err = module.run_command("%s stop --name=%s" % (vdocmd, vdoname))
+    rc, out, err = module.run_command([vdocmd, "stop", "--name=%s" % vdoname])
     if rc == 0:
         module.log("stopped VDO volume %s" % vdoname)
-
     return rc
 
 
 def activate_vdo(module, vdoname, vdocmd):
-    rc, out, err = module.run_command("%s activate --name=%s"
-                                      % (vdocmd, vdoname))
+    rc, out, err = module.run_command([vdocmd, "activate", "--name=%s" % vdoname])
     if rc == 0:
         module.log("activated VDO volume %s" % vdoname)
-
     return rc
 
 
 def deactivate_vdo(module, vdoname, vdocmd):
-    rc, out, err = module.run_command("%s deactivate --name=%s"
-                                      % (vdocmd, vdoname))
+    rc, out, err = module.run_command([vdocmd, "deactivate", "--name=%s" % vdoname])
     if rc == 0:
         module.log("deactivated VDO volume %s" % vdoname)
-
     return rc
 
 
 def add_vdooptions(params):
     vdocmdoptions = ""
     options = []
 
-    if ('logicalsize' in params) and (params['logicalsize'] is not None):
+    if params['logicalsize'] is not None:
```

Comment:
This (and the others) need to be reverted, since this function is also called with a reduced set of params: https://github.com/ansible-collections/community.general/blob/3f0ad52d14f42255c88fc3e5bacfb90325e55ae0/plugins/modules/system/vdo.py#L624

(You could also use `.get()` instead of `[]` though. That should also work.)

Reply:
Nice catch!